components/Container/index.tsx

```diff
@@ -0,0 +1,292 @@
+'use client'
+
+import React, {
+  CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+import {
+  Button,
+} from 'react-aria-components';
+
+import { DmpIcon } from '@/components/Icons';
+import {
+  useResponsive,
+  getSizeByName,
+} from '@/hooks/responsive';
+
+import './containers.scss';
+
+
+type DirectionType =
+  | null
+  | "left"
+  | "right";
+
+// Extend the CSSProps so that we can actually add css variables to tags.
+type CustomCSSProperties = CSSProperties & Record<string, string>;
+
+/**
+ * This is the base layout component. Our custom layout components should
+ * inherrit from this one. So that we have some common functionality for all
+ * layout containers.
+ */
+interface LayoutContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  style?: CustomCSSProperties,
+}
+
+export const LayoutContainer: React.FC<LayoutContainerProps> = ({
```

Comment:
Not a big deal, but it seems like it would be easier to find these components, if they were in their own files, with them all living in the "Container" directory?

Reply:
You are probably right. It's been in the back of my mind, but I was more concerned with just getting things done. Will pull these into their own files when I get the other changes done.

Reply:
Thanks Andre